Geometry/CaloGeometry/src/CaloSubdetectorGeometry.cc

```diff
@@ -40,51 +40,45 @@ CaloSubdetectorGeometry::getValidDetIds( DetId::Detector /*det*/    ,
    return m_validIds ;
 }
 
-const CaloCellGeometry* 
+std::shared_ptr<const CaloCellGeometry> 
 CaloSubdetectorGeometry::getGeometry( const DetId& id ) const
 {
-   return cellGeomPtr( CaloGenericDetId( id ).denseIndex() ) ;
+  return cellGeomPtr(CaloGenericDetId(id).denseIndex());
 }
 
 bool 
-CaloSubdetectorGeometry::present( const DetId& id ) const 
+CaloSubdetectorGeometry::present( const DetId& id ) const
 {
    return ( nullptr != getGeometry( id ) ) ;
```

Comment:
this method represents a significant cost fraction is all calls to 
CaloSubdetectorGeometry::getGeometry
here is a profile from reco
https://slava77sk.web.cern.ch/slava77sk/reco/cgi-bin/igprof-navigator/CMSSW_10_0_0_pre2-sign987.step3.136.831.100.pp/1098
I expect that in HLT context ::present actually dominates and may even be significant there.
The CPU cost of CaloSubdetectorGeometry::present is up by a factor of 2.5 primarily due to cycling through shared pointer creation and destruction.
I think that this extra cost can be avoided by e.g. renaming the old implementation of getGeometry to getGeometryRawPtr and keeping it here.

Also, it looks like this PR could be simplified substantially by keeping all old geometry method implementations where a raw pointer was returned, renaming them [perhaps also making private] with a RawPtr suffix and making all the new methods with old names returning shared_ptr created from a return value of RawPtr. This way we could reduce the cost of cycling through shared_ptr at least in the internal methods of geometry classes.